go-controller/pkg/level_driven_controller/level_driven_controller.go

```diff
@@ -0,0 +1,271 @@
+package level_driven_controller
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+const maxRetries = 15
+
+type ControllerConfig[T any] struct {
+	RateLimiter workqueue.RateLimiter
+	Informer    cache.SharedIndexInformer
+	Lister      func(selector labels.Selector) (ret []*T, err error)
+}
+
+type HandlerFuncs[T any] struct {
+	// ObjNeedsUpdate tells if object should be reconciled.
+	// May be called with oldObj = nil on Add, won't be called on Delete.
+	ObjNeedsUpdate func(oldObj, newObj *T) bool
+	Reconcile      func(key string) error
+	// InitialSync will be called on controller start with all existing objects.
+	// It ensures no events will be missed.
+	// If initial sync is not required, leave this function nil.
+	InitialSync func(objs []*T) error
+
+	// In case some events shouldn't be handled, we can skip them with this config.
+	// Add and Update are configured by one variable, because on restart all existing objects will trigger add event,
+	// which in reality may be an update copared to the previously handled events.
+	HandleAddUpdate bool
+	HandleDelete    bool
```

Comment:
it can, I just tried to save some hustle to queue and reconcile something that we know doesn't need to be reconciled (as you can just omit onDelete function with `AddHandler`)
I can remove it, or I can change it to `skipAdd/Update/Delete` so that default value is false, and you don't need to mention it unless you are trying to skip some event, wdyt?

Reply:
I would just remove it. API wise it complicates it seems unnecessarily if a user is still empowered to make those decisions in Reconcile.